### ISSUE TITLE
Upgrade Windows projects to VS 2019

### ISSUE THREAD

**Issue by ivan-mogilko** [adventuregamestudio/ags#2241]
VS 2015 free Community Edition is deprecated by Microsoft, and is no longer available for download from their official page (although there are still "unofficial" ways to acquire it).

At the same time the latest VS 2022 does not install on Windows 7, which I still use unfortunately. I might have to upgrade to Win10 or 11 soon, as more and more apps stop supporting it in their updates, but that will take some time.

Therefore I suggest to first move to VS 2019, as a temporary measure. According to [this Microsoft's page](https://learn.microsoft.com/en-us/lifecycle/products/visual-studio-2019) the VS 2019's lifecycle is scheduled to end in 2029.

~As we have a 3.6.1 coming towards a full stable release, I would not like to apply these changes to ags3 branch just yet. Instead I propose to begin with ags4 branch, and apply this change to ags3 only after 3.6.1 is fixed in a separate "release-3.6.1" branch.~ EDIT: 3.6.2 is still in beta, so theoretically we may switch it to 2019.

The changes to the project might include:
1. Upgrade the VS solutions and projects.
2. Fix any new compilation errors and outstanding warnings.
3. Switch to the newer "VC runtime" dependency, make certain that it runs on Windows 7 at least.
4. Replace VC runtime dependency in the AGS Editor's installer, readmes etc.
6. **Low priority:** check if it's still possible to build engine for Windows XP, using a separate configuration.

Task list, for keeping the record:
- [x] Update to VS 2019 in ags3 branch
- [x] Update to VS 2019 in ags4 branch
- [ ] Update projects and make/cmake scripts to C++14.

**Comment by ericoporto** [comment]
This has to be done in a PR and not with direct commits, as I believe after it's all done it won't build due to #1508

**Comment by ivan-mogilko** [comment]
I think we might as well move to at least to C++14 already. I would also consider C++17, because it has generic filesystem api.

**Comment by ericoporto** [comment]
C++14 is holdback for how old Linux we want to go. We are using Debian 8 (Jessie) and currently the stable release is Debian 12 (Bookworm).

**Comment by edmundito** [comment]
FWIW, I've been using 2022/C++ 17 with trivial adjustments and can contribute to the engine and editor without issues. So, the 2019 upgrade is OK with me.

**Comment by edmundito** [comment]
I pushed my changes for VS 2022 here for reference:
https://github.com/adventuregamestudio/ags/compare/master...edmundito:ags:edmundito/vs2022-compat

I just updated the latest version of 2022 (17.10.3) and had to:

* Specify all versions of C# to 8.0 (maybe the lowest compatible version it currently supports)
* Add an x86 as target (I just saw any CPU when I loaded)
* Let VS update the version info in the solution

**Comment by ivan-mogilko** [comment]
Upgraded in master branch (#2591).

Need to upgrade in ags4 now, and then preferably update C++ requirement to C++14 standard, to be able to use its feats without "compat" workarounds.

**Comment by ericoporto** [comment]
C++14 is the minimum VS2015 goes - I believe VS2022 the minimum is C++17. What actually is holding back the C++ standard version is the Linux builds, that is using a very old Linux distro - Debian Jessie.

**Comment by ivan-mogilko** [comment]
> What actually is holding back the C++ standard version is the Linux builds, that is using a very old Linux distro - Debian Jessie.

When you say "linux builds", do you mean the ones used by the AGS Editor when making linux game distributions?

How it is decided which debian distro is used there, can we update it?

**Comment by ivan-mogilko** [comment]
Hmm, there's still something wrong in ags4 branch, in "windows texts" job, concerning Compiler2 project. CI reports a path not found; I assume it can't find a hardcoded path to a Visual Studio to run it?

**Comment by ericoporto** [comment]
Looking into this

**Comment by ericoporto** [comment]
Fixed the ci in https://github.com/adventuregamestudio/ags/commit/476efb943271017c58b1a37c1cdf58cb176f1f32

**Comment by ivan-mogilko** [comment]
> Perhaps we could open a separate issue to discuss what we want to do regarding the Linux GCC support.

What is their state now?

**Comment by ericoporto** [comment]
We are using [Debian Jessie (Debian 8)](https://wiki.debian.org/DebianJessie) with [GCC version 4.9.2](https://gcc.gnu.org/onlinedocs/gcc-4.9.2/libstdc++/manual/manual/status.html#status.iso.2011).

Steam has a container called [sniper](https://gitlab.steamos.cloud/steamrt/steamrt/-/blob/steamrt/sniper/README.md#steam-runtime-3-sniper), based on Debian Bullseye (Debian 11), we could in theory upgrade to it, as long as gamedevs that release on Steam mark sniper in their Steamworks panel. Debian 11 comes with GCC 9.3 I think, which includes C++17.

If no container is used in Steam, any Linux distro that is more than 4 years old would probably not run these binaries though.

**Comment by ericoporto** [comment]
Just to mention, the Windows projects are already in C++14, MSVC doesn't support going lower since sometime now - I believe this was true before our current use of VS2022.
